### DISCUSSION TITLE
Self-hosted runners for autograding

### DISCUSSION THREAD

**Discussion by pinventado** [community/community#17796] [GitHub Education]
Are we able to use a self-hosted runner for auto-grading assignments? I have a lot of students in class and multiple assignments so I easily run out of credits.

**Comment by samus-aran** [comment]
Hi @pinventado, if you are a verified teacher using classroom we do offer increased minutes. Other community members ([Teachers within Global Campus](https://github.com/community/Global-Campus-Teachers/discussions/)) have commented on a [similar discussion](https://github.com/community/Global-Campus-Teachers/discussions/64). 

> if you have access to a server, you can [self-host](https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners) your own action runners. There is no charge from GitHub for self-hosted runners and you would have unlimited[1](https://github.com/community/Global-Campus-Teachers/discussions/64#user-content-fn-1-4557da6e3091d0a89a3af749ebaca749) minutes
1 you would still be limited by compute time and hours/minutes in a day and there might be [API and queue limits](https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners#usage-limits) [↩](https://github.com/community/Global-Campus-Teachers/discussions/64#user-content-fnref-1-4557da6e3091d0a89a3af749ebaca749)

**Reply by pinventado** [reply]
Unfortunately, the increased number of minutes is not enough. It is for a class I'm teaching on iOS app development. It requires a Mac server and the build times are pretty long. Multiplying that with the number of students and their commits, the increased hours are usually used up in a week or less :(
